src/Stratis.Bitcoin.Features.SmartContracts/PoA/WhitelistedHashChecker.cs

```diff
@@ -0,0 +1,40 @@
+using System.Collections.Generic;
+using NBitcoin;
+using Stratis.Bitcoin.Features.PoA.Voting;
+
+namespace Stratis.Bitcoin.Features.SmartContracts.PoA
+{
+    /// <summary>
+    /// Checks hashes against a whitelist.
+    /// </summary>
+    public class WhitelistedHashChecker : IWhitelistedHashChecker
+    {
+        private readonly IWhitelistedHashesRepository whitelistedHashesRepository;
+
+        public WhitelistedHashChecker(IWhitelistedHashesRepository whitelistedHashesRepository)
+        {
+            this.whitelistedHashesRepository = whitelistedHashesRepository;
+        }
+
+        /// <summary>
+        /// Checks that a supplied hash is present in the whitelisted hashes repository.
+        /// </summary>
+        /// <param name="hash">The bytes of the hash to check.</param>
+        /// <returns>True if the hash was found in the whitelisted hashes repository.</returns>
+        public bool CheckHashWhitelisted(byte[] hash)
+        {
+            List<uint256> allowedHashes = this.whitelistedHashesRepository.GetHashes();
+
+            if (hash.Length != 32)
```

Comment:
Move this above the gethashes (micro optimisation but could be a new List{200} items we get to avoid)